frontend/src/metabase/visualizations/lib/scalarUtils.js

```diff
@@ -0,0 +1,26 @@
+import { formatValue } from "metabase/lib/formatting";
+
+// used below to determine whether we show compact formatting
+const COMPACT_MAX_WIDTH = 250;
+const COMPACT_WIDTH_PER_DIGIT = 25;
+const COMPACT_MIN_LENGTH = 6;
+
+export function compactifyValue({ formatOptions, value, width }) {
```

Comment:
What a nice little function! Extracting this little puppy is a really awesome opportunity to

1) convert it to typescript, and
2) add some unit tests

😁